File: evo-aspnetcore/evo-aspnetcore/Models/Funcionario.cs
using System.ComponentModel.DataAnnotations;

namespace EvoTest.Models
{
    public class Funcionario
    {
        [Key]
        public int? Id { get; set; }
        public string? Nome { get; set; }
        public string? Foto{ get; set; }    
        public string? RG { get; set; }
        public Departamento? Departamento { get; set; }
        public int? DepartamentoId { get; set; }

        public Funcionario() { }
        public Funcionario(int id, string nome, string foto, string rg, int departamentoId)
        {
            this.Id = id;
            this.Nome = nome;
            this.Foto = foto;
            this.RG = rg;
            this.DepartamentoId = departamentoId;
        }
    }
}

File: evo-angular/src/app/models/funcionario.ts
import { departamento } from "./departamento";

export class funcionario {

    constructor() {

        this.id = 0;
        this.nome = '';
        this.foto = '';
        this.rg = '';
        this.departamento = null; 
        this.departamentoId = 0
        
    }

    id: number;
    nome: string;
    foto: string;
    rg: string;
    departamento: departamento | null;
    departamentoId: number | null;
  }